src/pages/api/iexec/cron/new-proposal.tsx

```diff
@@ -0,0 +1,62 @@
+import * as cron from 'node-cron';
+import mongoose from 'mongoose';
+import { Proposal } from './proposal-model';
+import { getProposalsFromPlatformServices } from '../../../../queries/proposals';
+import { IProposal } from '../../../../types';
+import { sendMailToMyContacts } from '../../utils/sendMailToMyContacts';
+import { userGaveAccessToPlatform } from '../../utils/iexec-utils';
+
+const setCron = async () => {
+  const mongoUri = process.env.NEXT_PUBLIC_MONGO_URI;
+
+  if (!mongoUri) {
+    throw new Error('MongoDb URI is not set');
+  }
+  await mongoose.connect(mongoUri as string);
+
+  cron.schedule('0 0 */1 * * *', async () => {
+    console.log('Running a task every 1 hour');
+    const platformId = process.env.NEXT_PUBLIC_PLATFORM_ID;
+
+    if (!platformId) {
+      throw new Error('Private key is not set');
+    }
+    try {
+      const response = await getProposalsFromPlatformServices(platformId);
+      const nonSentProposalIds: IProposal[] = [];
+
+      // Check if some proposals are not already in the DB
+      if (response.data.data.proposals.length > 0) {
+        for (const proposal of response.data.data.proposals as IProposal[]) {
+          const existingProposal = await Proposal.findOne({ id: proposal.id });
+          if (!existingProposal) {
+            nonSentProposalIds.push(proposal);
+          }
+        }
+      }
+
+      // If some proposals are not already in the DB, email the hirer & persist the proposal in the DB
+      if (nonSentProposalIds) {
+        for (const proposal of nonSentProposalIds) {
+          // Check if user granted access to his email
+          const userGaveAccess = await userGaveAccessToPlatform(proposal.service.buyer.address);
+          if (!userGaveAccess) {
+            console.warn('User did not grant access to his email');
+            continue;
+          }
+
+          await sendMailToMyContacts(
+            'You got a new proposal !',
+            `You just received a new proposal from the service ${proposal.service.id} you posted on TalentLayer !`,
```

Comment:
To add: name of the seller, rateAmount and rateToken, name of the platform, URL to the proposal, proposal title, proposal desc

Reply:
FYI: "Title" is not mapped in the graph